module/datamanager.py

```diff
@@ -1000,21 +1024,42 @@ def get_overall_it_state(self, user):
         return hosts_state, services_state
 
     def get_important_elements(self, user, type='all', sorter=worse_first):
-        return self.search_hosts_and_services('bi:>2 ack:false type:%s' % type, user=user, sorter=sorter)
+        # BI is greater than the minimum business impact
+        # todo: @maethor: why searching with ack:false ? an element is important whether it is ack or not...
+        search = "bi:>%d " % self.min_business_impact + 'ack:false type:%s' % type
+        return self.search_hosts_and_services(search, user=user, sorter=sorter)
 
-    def get_impacts(self, user, search='is:impact bi:>=0 type:all', sorter=worse_first):
+    def get_impacts(self, user, search='is:impact type:all', sorter=worse_first):
         if "is:impact" not in search:
             search = "is:impact " + search
+        if "bi:" not in search:
+            search = "bi:>=%d " % self.min_business_impact + search
         return self.search_hosts_and_services(search, user=user, get_impacts=True, sorter=sorter)
 
-    def get_problems(self, user, search='isnot:UP isnot:OK isnot:PENDING bi:>=0 type:all',
-                     get_acknowledged=False, get_downtimed=False, sorter=worse_first):
+    def get_problems(self, user, search='isnot:UP isnot:OK isnot:PENDING isnot:UNKNOWN isnot:ACK isnot:DOWNTIME '
+                                        'is:HARD type:all', get_acknowledged=False, get_downtimed=False,
+                     sorter=worse_first):
+        """What are problems:
+        - hosts/services (type:all) that are not OK/UP in a HARD state (isnot:hUP isnot:hOK)
+        - and that are not PENDING or UNKNOWN
+        - and which business impact is greater than or equel the minimum business impact (defaults 2)
+        """
         if "isnot:UP" not in search:
             search = "isnot:UP " + search
         if "isnot:OK" not in search:
             search = "isnot:OK " + search
         if "isnot:PENDING" not in search:
             search = "isnot:PENDING " + search
+        if "isnot:UNKNOWN" not in search:
+            search = "isnot:UNKNOWN " + search
+        if "isnot:ACK" not in search:
+            search = "isnot:ACK " + search
+        if "isnot:DOWNTIME" not in search:
+            search = "isnot:DOWNTIME " + search
+        if "is:HARD" not in search:
+            search = "is:HARD" + search
+        if "bi:" not in search:
+            search = "bi:>=%d " % self.min_business_impact + search
```

Comment:
NO! Why are you doing all this!?! No!

Reply:
The*get_problems* function is only used in the Wall view and it needs to be consistent with all the UI ... why forcing only 3 criteria whereas more are necessary to qualify a problem ...

Reply:
OK.